integration-tests/tests/aws/kbv-check-happy.test.ts

```diff
@@ -0,0 +1,67 @@
+import { stackOutputs } from "./resources/cloudformation-helper";
+import { clearItems, populateTable } from "./resources/dynamodb-helper";
+import { executeStepFunction } from "./resources/stepfunction-helper";
+
+describe("HMRC KBV Check ", () => {
+  const stateMachineInput = {
+    sessionId: "12345",
+  };
+  const testUser = {
+    sessionId: "12345",
+    nino: "AA000003D",
+  };
+
+  const testQuestions = [
+    {
+      sessionId: "12345",
+      questionKey: "ita-bankaccount",
+      answered: "false",
+      correlationId: "93dcc67c-fe6d-4bd7-b68f-2bd848e0d572",
+      info: new Map(),
+    },
+    {
+      sessionId: "12345",
+      questionKey: "ita-bankaccount2",
+      answered: "false",
+      correlationId: "93dcc67c-fe6d-4bd7-b68f-2bd848e0d573",
+      info: new Map(),
+    },
+  ];
+
+  let output: Partial<{
+    PersonalIdenityTable: string;
+    QuestionsTable: string;
+    QuestionStateMachineArn: string;
+  }>;
+  beforeEach(async () => {
+    output = await stackOutputs(process.env.STACK_NAME);
+    await populateTable(testUser, output.PersonalIdenityTable);
+    testQuestions.forEach(
+      async (question) => await populateTable(question, output.QuestionsTable)
+    );
+  });
+
+  afterEach(async () => {
+    await clearItems(output.PersonalIdenityTable as string, {
+      sessionId: testUser.sessionId,
+    });
+    testQuestions.forEach(
+      async (question) =>
+        await clearItems(output.QuestionsTable as string, {
```

Comment:
This loop needs to be changed because the promise returned by clearItems is currently ignored. As a result, the forEach loop iterates over all the items without waiting for the results of clearItems. This can potentially cause unexpected issues later on. To fix this, I recommend changing it to a regular for...of loop, like this:

```
for (const question of testQuestions) {
  await clearItems(output.QuestionsTable as string, {
    sessionId: question.sessionId,
    questionKey: question.questionKey,
  });
}
```

Alternatively, a better solution might be to use `Promise.all` which is used to wait all the asynchronous operations like so:

```
await Promise.all(
  testQuestions.map(async (question) =>
    clearItems(output.QuestionsTable as string, {
      sessionId: question.sessionId,
      questionKey: question.questionKey,
    })
  )
);
```

Reply:
Do the same for all the areas in the code were the promises are ignored (your IDE should give a warning)